server.js

```diff
@@ -72,9 +73,17 @@ http.listen(port, () => console.log(`Server listening on port ${port}`));
 
 // TODO Define an Observable that maps processed actions of type 'holdRoom'
 // to FSAs of type setOccupancy (which will be sent out to clients)
+const agent = new Agent();
 
-// TODO Process holdRoom actions through the store so new clients
-// will get the actual state. Later, we'll persist the change in the db
+const roomHoldOccupancyChanges = agent.allOfType("holdRoom").pipe(
+  map(action => ({
+    type: "setOccupancy",
+    payload: {
+      num: action.payload.num,
+      occupancy: action.payload.hold ? "hold" : "open"
+    }
+  }))
+);
```

Comment:
You can define an Observable out of a filtered down set of actions that the `agent` processes using `agent.allOfType`. Here we provide an Observable of `setOccupancy` actions, in correspondence with the `holdRoom` actions. This simulates an approval process, and keeps the flow where clients request via `holdRoom`, and servers communicate via `setOccupancy`.